platform-hub-api/app/services/kubernetes/static_token_service.rb

```diff
@@ -0,0 +1,82 @@
+module Kubernetes
+  module StaticTokenService
+    extend self
+
+    # Static tokens are stored in HashRecord as:
+    # data: [
+    #   {"token" => "token1", user" => "user1", "uid" => "uid1", "groups" => ["group1","group2"]}
+    #   {"token" => "token2", user" => "user2", "uid" => "uid2", "groups" => []}
+    #   ...
+    # ]
+    # Note: All tokens are encrypted!
+
+    def create_or_update(cluster, kind, user_name, groups = [])
+      static_tokens = get_static_tokens(cluster, kind)
+
+      record = static_tokens.data.find do |t|
+        t['user'] == user_name.to_s
+      end
+
+      if record.nil?
+        token = Kubernetes::TokenService.generate_secure_random
+        record = Hashie::Mash.new(
+          token: ENCRYPTOR.encrypt(token),
+          user: user_name,
+          uid: Kubernetes::TokenService.generate_secure_random,
+          groups: Kubernetes::TokenService.send(:cleanup, groups)
+        )
+
+        static_tokens.data << record
+      else
+        record['groups'] = groups.present? ? 
+          Kubernetes::TokenService.send(:cleanup, groups) : []
+      end
+
+      static_tokens.save!
+      "Created/updated #{kind} account for `#{user_name}` (token: #{ENCRYPTOR.decrypt(record['token'])})"
+    end
+
+    def delete_by_user_name(cluster, kind, user_name)
+      static_tokens = get_static_tokens(cluster, kind)
+
+      static_tokens.data.reject! do |t|
+        t['user'] == user_name.to_s
+      end
+
+      static_tokens.save!
+      "Deleted #{kind} account for `#{user_name.to_s}`"
+    end
+
+    def delete_by_token(cluster, kind, token)
+      static_tokens = get_static_tokens(cluster, kind)
+
+      static_tokens.data.reject! do |t|
+        ENCRYPTOR.decrypt(t['token']) == token
+      end
+
+      static_tokens.save!
+      "Deleted #{kind} account with token `#{token.to_s}`"
+    end
```

Comment:
With operations like these, we should really consider pessimistic table level locking? As it's possible that two users claim tokens at the same time and cause data inconsistency.

(Note: I tried using Postgres' advisory locks in https://github.com/UKHomeOffice/platform-hub/pull/59 but they didn't seem to be working as expected in our set up, so avoid those for now, I think).